### PR TITLE
feat: [AB#17839] implement static site 404 page with routing

### PR DESCRIPTION
**Pull request by hinzed1127**
## Description

Adds a 404 page to the static site, replacing the Next.js default 404.

### Ticket

- This pull request resolves [#17839](https://dev.azure.com/NJInnovation/Business%20First%20Stop/_workitems/edit/17839).
- Figma design can be found [here](https://www.figma.com/design/pk0neG436QNTm805O603Vp/96-100-Business-Experience-Sprint-Designs?node-id=70742-6033&t=YQXo1SN7NHn9bsIS-4).

### Approach

- The new 404 page, `app/[locale]/not-found.tsx`, follows the guidance of [[Next.js documentation](https://nextjs.org/docs/app/api-reference/file-conventions/not-found)](https://nextjs.org/docs/app/api-reference/file-conventions/not-found). Any framework-level considerations are commented in the code for clarity.
- In testing the 404 page, a bug in semantic header structure for the footer was surfaced, changing some `h3` elements to `h2`.
- Added the new page's copy to both locale message files and moved the 404 svg into the static assets.

### Steps to Test

1. `cd packages/static-site && pnpm dev`
2. Visit `http://localhost:4000/some-nonexistent-route` 
3. `pnpm lint && pnpm typecheck && pnpm test && pnpm test:accessibility && pnpm build` should all pass.
4. Post-merge validation in dev: Clicking "Chat with a NJ-based business expert" should open the intercom chat window

### Notes

- The root `app/not-found.tsx` fallback is intentionally minimal and untranslated. Real users should never see it. It only catches paths the middleware never routes into `[locale]` at all. It follows the recommended pattern of the `next-intl` docs.

## Code author checklist

- [x] I have rebased this branch from the latest main branch
- [x] I have performed a self-review of my code
- [x] My code follows the style guide
- [x] I have created and/or updated relevant documentation on the engineering documentation website
- [x] I have not used any relative imports
- [x] I have pruned any instances of unused code
- [x] I have not added any markdown to labels, titles and button text in config
- [x] ~~If I added/updated any values in `userData` (including `profileData`, `formationData` etc), then I added a new migration file~~ n/a
- [x] I have checked for and removed instances of unused config from CMS
- [x] If I added any new collections to the CMS config, then I updated the search tool and `cmsCollections.ts` (see CMS Additions in Engineering Reference/FAQ on the engineering documentation site)
- [x] ~~I have updated relevant `.env` values in `.env-template`, in Bitwarden, and in our workspaces~~ n/a
- [x] I have added the `pr-show` or `pr-review` label on GitHub to show or request reviews

### CODE REVIEW

File: packages/static-site/public/img/404-page-not-found-icon.svg
Comment (hinzed1127) [inline packages/static-site/public/img/404-page-not-found-icon.svg:1]: while similar, this is _slightly_ different in appearance from the `web/` 404 icon, so no worries for overlap

File: packages/static-site/AGENTS.md [hunk 1]
```diff
@@ -35,6 +35,11 @@ pnpm test:accessibility  # Playwright + axe-core a11y tests (requires running se
 - **Do not use Jest APIs here.** Use `vi.fn()`, `vi.mock()`, etc. (Vitest, not Jest).
 - **Do not use ESLint or Prettier.** Biome handles both linting and formatting.
 - Localized strings live in `messages/*.json`. Do not hardcode user-facing copy in components.
+- **Tests must not hardcode user-facing copy.** When a query needs matchable text — `getByText`,
+  or the `name` field inside `getByRole`/`getByLabelText`/etc. — pass the value from
+  `getApplicationMessages({ locale })` (or other real config), not a retyped copy of the string.
+  Hardcoded copies silently drift from `messages/*.json` and pass even after the real copy
+  changes. See `app/[locale]/not-found.test.tsx` and `components/learn/LearnSideNav.test.tsx`.
```
Comment (hinzed1127) [inline packages/static-site/AGENTS.md:42]: an attempted harness improvement, since I've noticed most generated tests have this issue if you don't instruct against it

File: web/AGENTS.md [hunk 2]
```diff
@@ -67,3 +67,12 @@ services (`yarn services:up`) and a running dev server.
...
 Path aliases are configured in `tsconfig.json`. Use `@/components/...`, `@/lib/...`, etc. rather than relative paths crossing directory boundaries.
+
+## Test Conventions
+
+**Tests must not hardcode user-facing copy.** When a query needs matchable text —
+`getByText`, or the `name` field inside `getByRole`/`getByLabelText`/etc. — pass the
+value from `getMergedConfig()` (`@businessnjgovnavigator/shared/contexts`), not a
+retyped copy of the string. Hardcoded copies silently drift from the real content
+and pass even after the copy changes. See `NavbarBusinessNjGovLogo.test.tsx` or
+`ChecklistTag.test.tsx` for the pattern.
```
Comment (hinzed1127) [inline web/AGENTS.md:78]: see previous comment

File: packages/static-site/app/[locale]/[...rest]/page.tsx [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+/**
+ * Catches every otherwise-unmatched path under a `[locale]` prefix.
+ *
+ * Without this, Next.js never matches (and therefore never renders) a route
+ * component for an unknown deep path, so `notFound()` is never called from
+ * within the locale segment and the request falls through to the untranslated
+ * root `not-found.tsx` instead of the branded `[locale]/not-found.tsx`.
+ * https://next-intl.dev/docs/environments/error-files
+ */
```
Comment (hinzed1127) [inline packages/static-site/app/[locale]/[...rest]/page.tsx:11]: comments like this and the one in `packages/static-site/app/not-found.tsx` are the kind I would argue add more value: they're rooted in specifics to a framework and not obvious at a glance unless you know the conventions of Next.js while. You could probably piece it together with the names, but this makes it more explicit.

File: packages/static-site/app/globals.css [hunk 4]
```diff
@@ -64,6 +85,18 @@ body:has(.layout-wide) .usa-nav__inner {
...
 }
 
+/*
+ * The footer's "Return to top" link is a no-op on the 404 page — there is no
+ * scrollable content above it to return to. The footer is a sibling of
+ * `<main>`, not an ancestor of `.page-not-found`, so this is scoped via
+ * `body:has()` rather than `.grid-container`/`.grid-row` scoping, matching
+ * the `.layout-wide` pattern above (the layout can't see which child route
+ * is rendering).
+ */
```
Comment (hinzed1127) [inline packages/static-site/app/globals.css:95]: These comments I'm less strongly in favor of, but they're still catching edge cases that are more subtle, so I don't really see harm in offering the additional context this way

**Review by hinzed1127 — COMMENTED** [review]
Self-review. Current test suite failure is a flakey test in `web/`, shouldn't be a blocker to these changes
